### PR TITLE
feat: poem-web integration

### PR DESCRIPTION
**Pull request by oddgrd**
### Changes

- [x] Implement `Service` for Poem Web apps.
- [x] Support `cargo shuttle init --poem`.
- [x] Add `hello-world` example.
- [x] Add `postgres` example.
- [x] Add tests for the new examples.
- [x] Update CI. 

### How I tested my changes

I added e2e integration tests for the new examples.
I added integration tests in `cargo-shuttle` for the new examples.
I added a `cargo shuttle init` unit test and updated the `cargo shuttle init` integration test.
I deployed with `cargo run --manifest-path ../../../Cargo.toml --bin cargo-shuttle -- deploy` and verified with `curl` that it worked correctly.

### Relevant Issues

Fixes issue #269.
Partially fixes issue #263.

### CODE REVIEW

File: cargo-shuttle/src/init.rs [hunk 1]
```diff
@@ -155,6 +155,65 @@ impl ShuttleInit for ShuttleInitTide {
...
+        use poem::{get, handler, middleware::Tracing, EndpointExt, Route};
+
+        #[handler]
+        fn hello_world() -> &'static str {
+            "Hello, world!"
+        }
+
+        #[shuttle_service::main]
+        async fn poem() -> shuttle_service::ShuttlePoem<impl poem::Endpoint> {
+            let _ = tracing_subscriber::fmt().try_init();
```
Comment (chesedo) [inline cargo-shuttle/src/init.rs:208]: This is a great addition! We should probably bake this into shuttle though, which means a separate project (#280) :sweat_smile: Will this PR be complete if we remove this line?

**Review by chesedo — COMMENTED** [review]
This is looking great, thanks @oddgrd! I am happy to merge, just want to confirm the `tracing` part.

File: cargo-shuttle/src/init.rs (hunk 1 above)
Comment (oddgrd) [reply cargo-shuttle/src/init.rs:208]: Thanks for the feedback! That makes sense. Should I just roll back the tracing commit? It'll be ready to go then.

Comment (chesedo) [reply cargo-shuttle/src/init.rs:208]: That would be great, thanks

**Review by chesedo — APPROVED** [review]
Nice :star_struck:, thanks @oddgrd

**Pull request merged**